packages/react/src/experimental/OneDataCollection/index.tsx

```diff
@@ -354,6 +360,47 @@ export const OneDataCollection = <
 
   const [totalItems, setTotalItems] = useState<undefined | number>(undefined)
 
+  const { emptyState, setEmptyStateType } = useEmptyState(emptyStates, {
+    retry: () => {
+      setEmptyStateType(false)
+      setCurrentFilters({ ...currentFilters })
+    },
+    clearFilters: () => {
+      setEmptyStateType(false)
+      setCurrentFilters({})
+    },
+  })
+
+  const onLoadData = ({
+    totalItems,
+    filters,
+    isInitialLoading,
+    search,
+  }: Parameters<OnLoadDataCallback<Record, Filters>>[0]) => {
+    if (isInitialLoading) return
+
+    setTotalItems(totalItems)
+    setEmptyStateType(
+      totalItems === 0
+        ? Object.keys(filters).length > 0 || search
+          ? "no-results"
+          : "no-data"
+        : false
+    )
```

Comment:
Would you consider to create a const with Kind of EmptyState and set a function sach as:
```
const EmptyState = {
  NO_RESULTS: "no-results",
  NO_DATA: "no-data",
  NONE: false
};

function getEmptyStateType(totalItems, filters, search) {
  if (totalItems > 0) return EmptyState.NONE;
  if (Object.keys(filters).length > 0 || search) return EmptyState.NO_RESULTS;
  return EmptyState.NO_DATA;
}
```

Then you can use this function in the `setEmptyStateType`
`setEmptyStateType(getEmptyStateType(totalItems, filters, search));`

Reply:
this is a kind of fake enum, i prefer to use an union type based in a const array
```
export const emptyStatesTypes = ["no-data", "no-results", "error"]
export type EmptyStateType = (typeof emptyStatesTypes)[number]
```

It's safe enough

Regarding to the function, i guess is not worthy in terms of reusabiliy (as this will be only used here), but it will refactor for readabiliy